src/xdr/XdrEncoder.ts

```diff
@@ -0,0 +1,295 @@
+    writer.move(8);
+  }
+
+  /**
+   * Writes an XDR unsigned 64-bit integer (unsigned hyper) in big-endian format.
+   */
+  public writeUnsignedHyper(uhyper: number | bigint): void {
+    const writer = this.writer;
+    writer.ensureCapacity(8);
+
+    if (typeof uhyper === 'bigint') {
+      // Convert bigint to two 32-bit values for big-endian encoding
+      const high = Number((uhyper >> BigInt(32)) & BigInt(0xffffffff));
+      const low = Number(uhyper & BigInt(0xffffffff));
+      writer.view.setUint32(writer.x, high, false); // high 32 bits
+      writer.view.setUint32(writer.x + 4, low, false); // low 32 bits
+    } else {
+      const truncated = Math.trunc(Math.abs(uhyper));
+      const high = Math.floor(truncated / 0x100000000);
+      const low = truncated >>> 0;
+      writer.view.setUint32(writer.x, high, false); // high 32 bits
+      writer.view.setUint32(writer.x + 4, low, false); // low 32 bits
+    }
+    writer.move(8);
+  }
+
+  /**
+   * Writes an XDR float value using IEEE 754 single-precision in big-endian format.
+   */
+  public writeFloat(float: number): void {
+    const writer = this.writer;
+    writer.ensureCapacity(4);
+    writer.view.setFloat32(writer.x, float, false); // big-endian
+    writer.move(4);
+  }
+
+  /**
+   * Writes an XDR double value using IEEE 754 double-precision in big-endian format.
+   */
+  public writeDouble(double: number): void {
+    const writer = this.writer;
+    writer.ensureCapacity(8);
+    writer.view.setFloat64(writer.x, double, false); // big-endian
+    writer.move(8);
+  }
+
+  /**
+   * Writes an XDR quadruple value (128-bit float).
+   * Note: JavaScript doesn't have native 128-bit float support.
+   */
+  public writeQuadruple(quad: number): void {
+    throw new Error('not implemented');
+  }
+
+  /**
+   * Writes XDR opaque data with fixed length.
+   * Data is padded to 4-byte boundary.
+   */
+  public writeOpaque(data: Uint8Array, size: number): void {
+    if (data.length !== size) {
+      throw new Error(`Opaque data length ${data.length} does not match expected size ${size}`);
+    }
+
+    const writer = this.writer;
+    const paddedSize = Math.ceil(size / 4) * 4;
+    writer.ensureCapacity(paddedSize);
+
+    // Write data
+    writer.buf(data, size);
+
+    // Write padding bytes
+    const padding = paddedSize - size;
+    for (let i = 0; i < padding; i++) {
+      writer.u8(0);
+    }
+  }
+
+  /**
+   * Writes XDR variable-length opaque data.
+   * Length is written first, followed by data padded to 4-byte boundary.
+   */
+  public writeVarlenOpaque(data: Uint8Array): void {
+    this.writeUnsignedInt(data.length);
+
+    const writer = this.writer;
+    const paddedSize = Math.ceil(data.length / 4) * 4;
+    writer.ensureCapacity(paddedSize);
+
+    // Write data
+    writer.buf(data, data.length);
+
+    // Write padding bytes
+    const padding = paddedSize - data.length;
+    for (let i = 0; i < padding; i++) {
+      writer.u8(0);
+    }
+  }
```

Comment:
Reuse `.writeOpaque()` here if possible:


```suggestion
    this.writeOpaque(data);
  }
```